components/core/src/clp/utils.hpp

```diff
@@ -60,23 +60,10 @@ namespace clp {
     bool validate_paths_exist (const std::vector<std::string>& paths);
 
     /**
-     * Creates and opens a file within the given archive
+     * Closes the encoded file in the given archive and marks it ready for a segment
      * @param archive
-     * @param path_for_compression
-     * @param group_id
-     * @param orig_file_id
-     * @param split_ix
-     * @return A pointer to the file
-     */
-    streaming_archive::writer::File* create_and_open_file (streaming_archive::writer::Archive& archive, const std::string& path_for_compression,
-                                                           group_id_t group_id, const boost::uuids::uuid& orig_file_id, size_t split_ix);
-
-    /**
-     * Closes the given encoded file and marks it ready for a segment
-     * @param archive
-     * @param file
      */
-    void close_file_and_mark_ready_for_segment (streaming_archive::writer::Archive& archive, streaming_archive::writer::File*& file);
+    void close_file_and_mark_ready_for_segment (streaming_archive::writer::Archive& archive);
```

Comment:
* How about `close_file_and_append_to_segment`? Marking files ready for a segment is from an older version of CLP where we would accumulate files before adding them to a segment.
* Make sure to change the comment if you go with the change above.